interface/src/app/gene/searchMany/searchMany.js

```diff
@@ -25,6 +25,15 @@ angular.module('adage.gene.searchMany', [
               // be obtained from the ML model. This is the same as
               // the issue in geneSearchForm (also with $scope.organism).
               $scope.organism = 'Pseudomonas aeruginosa';
+
+              // '$scope.autocomplete.autocomlete' holds a boolean value,
+              // of whether or not the user wants to use autocomplete
+              // search to look up a few genes. If not, then the autocomplete
+              // search panel will disappear, and the panel to search many
+              // genes will appear. We need to send the value itself inside
+              // an object so that it gets sent properly back and forth
+              // through the corresponding $scopes.
+              $scope.autocomplete = {autocomplete: true};
```

Comment:
Aha! Thank you @dongbohu for the documentation links you provided on Slack - this is all much more clear now. This doc (https://github.com/angular/angular.js/wiki/Understanding-Scopes) and its link to the video of Misko Hevery on best practices (http://www.youtube.com/watch?v=ZhfUv0spHCY&feature=youtu.be&t=30m) were particularly useful in understanding how I was actually creating a new property in the directive's childScope instead of referring to the desired property in the parentScope and changing it from true to false and back. With this I am now following the best practice of "always have a '.' in your ng-models" (see video). This is something that we should keep in mind going forward and should change in the existing code whenever we see it happening.